source/developers-guide/address-management-guide/index.md

```diff
@@ -6,52 +6,245 @@ shopware_version: 5.2.0
+    $builder = $event->getBuilder();
+
+    $builder->get('additional')
+            ->add('neighboursName', \Symfony\Component\Form\Extension\Core\Type\TextType::class, [
+                'constraints' => [
+                    new \Symfony\Component\Validator\Constraints\NotBlank()
+                ]
+            ])
+            ->add('neighboursEmail', \Symfony\Component\Form\Extension\Core\Type\EmailType::class, [
+                'constraints' => [
+                    new \Symfony\Component\Validator\Constraints\NotBlank(),
+                    new \Symfony\Component\Validator\Constraints\Email()
+                ]
+            ])
+            ->add('neighboursPhone', \Symfony\Component\Form\Extension\Core\Type\IntegerType::class);
+});
+```
+
+#### Example #3: Handle the additional data in the AddressService
+
+This example will decorate the address service in order to handle additional data. First, we need a new class, which implements the `AddressServiceInterface`.
+
+##### MyAddressService.php
+
+```php
+<?php
+
+use Shopware\Bundle\AccountBundle\Service\AddressServiceInterface;
+use Shopware\Models\Customer\Address;
+use Shopware\Models\Customer\Customer;
+
+class MyAddressService implements AddressServiceInterface
+{
+    /**
+     * @var AddressServiceInterface
+     */
+    private $coreAddressService;
+
+    public function __construct(AddressServiceInterface $coreAddressService)
+    {
+        $this->coreAddressService = $coreAddressService;
+    }
+
+    /**
+     * Handles additional data
+     * @param Address $address
+     */
+    private function handleAdditionalData($address)
+    {
+        // handle additional data
+        $additional = $address->getAdditional();
+
+        if (!empty($additional['neighboursEmail'])) {
+            mail($additional['neighboursEmail'], 'Address changed', 'Nice mail text');
+        }
+    }
+
+    public function create(Address $address, Customer $customer)
+    {
+        $this->coreAddressService->create($address, $customer);
+        $this->handleAdditionalData($address);
+
+    }
+
+    public function update(Address $address)
+    {
+        $this->coreAddressService->update($address);
+        $this->handleAdditionalData($address);
+    }
+
+    public function delete(Address $address)
+    {
+        $this->coreAddressService->delete($address);
+    }
+
+    public function isValid(Address $address)
+    {
+        $this->coreAddressService->isValid($address);
+    }
+
+    public function setDefaultBillingAddress(Address $address)
+    {
+        $this->coreAddressService->setDefaultBillingAddress($address);
+    }
+
+    public function setDefaultShippingAddress(Address $address)
+    {
+        $this->coreAddressService->setDefaultShippingAddress($address);
+    }
+}
+```
+
+Now that we have created the class, we need to replace the existing service with our new service. For this, subscribe to a new event in your `Bootstrap.php` file.
```

Comment:
I would not write "replace" but "decorate" 
as the service gets not really replaced. Replace and Decorate are two different things